### PR TITLE
chore: use the github reusable workflow to create jira issue

### PR DESCRIPTION
**Pull request by yuhao900914**
### Checklist

* [ ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Kotlin/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:

### CODE REVIEW

**Review by liuyang1520 — APPROVED** [review]

File: .github/workflows/jira-issue-create.yml [hunk 1]
```diff
@@ -1,39 +1,21 @@
...
-            "customfield_10006": 1391,
-            "customfield_11481": {"value": "Governance 🕵️"},
-            "customfield_11200": {"value": "Developer Experience"},
-            "labels": ["kotlin"]
-          }' # sprint, pillar, pod, labels
-
-      - name: Log created issue
-        run: echo "Issue AMP-${{ steps.create.outputs.issue }} was created"
+  call-workflow-passing-data:
+    uses: amplitude/Amplitude-TypeScript/.github/workflows/jira-issue-create-template.yml@main
```
Comment (qingzhuozhen) [inline .github/workflows/jira-issue-create.yml:14]: Do we have plan to use DXOC in the future for triage? Or we can keep this for now?

**Review by qingzhuozhen — APPROVED** [review]

File: .github/workflows/jira-issue-create.yml (hunk 1 above)
Comment (yuhao900914) [reply .github/workflows/jira-issue-create.yml:14]: I think no matter if we want to use DXOC or not, with this reusable workflow, we only need to update the template located in amplitude-typescript. Don't need to update the workflow one by one.

Comment (yuhao900914) [reply .github/workflows/jira-issue-create.yml:14]: Oh, also need to update the project secret in each repo for now.

Comment (qingzhuozhen) [reply .github/workflows/jira-issue-create.yml:14]: yeah, just wonder if we need an extra field besides label. Maybe we can keep what we have for now and reuse it.

**Pull request merged**